src/lib/core/components/computations/ComputeJobStatusHook.ts

```diff
@@ -0,0 +1,101 @@
+import { useCallback, useEffect, useRef, useState } from 'react';
+import { JobStatusReponse } from '../../api/ComputeClient';
+import { useComputeClient, useStudyMetadata } from '../../hooks/workspace';
+import { Analysis, NewAnalysis } from '../../types/analysis';
+import { Computation } from '../../types/visualization';
+
+export type JobStatus = JobStatusReponse['status'] | 'requesting';
+
+/**
+ * Polls the compute service for the status of a compute's job
+ */
+export function useComputeJobStatus(
+  analysis: Analysis | NewAnalysis,
+  computation: Computation,
+  computeName?: string
+) {
+  const [jobStatus, setJobStatus] = useState<JobStatus>();
+  const computeClient = useComputeClient();
+  const studyMetadata = useStudyMetadata();
+  const sharedStatusRef = useRef<JobStatus>();
```

Comment:
So we "mirror" the `jobStatus` value using a ref because we need to use it inside the useEffect, but we can't use it directly or it would have to go in the dependencies, and that would cause some circular issues that I can't quite put into words?

Reply:
Another comment needed! Yes, your description is correct. Basically the effect would run on every status change, which is not what we want.

The idea is that we want to run the loop until the status is terminal.